e2e/cloud-plugins-suite/azure-monitor.spec.ts

```diff
@@ -0,0 +1,297 @@
+  e2e.components.Tab.title('Variables').click();
+  if (isFirst) {
+    e2e.pages.Dashboard.Settings.Variables.List.addVariableCTAV2().click();
+  } else {
+    e2e.pages.Dashboard.Settings.Variables.List.newButton().click();
+  }
+  e2e.pages.Dashboard.Settings.Variables.Edit.General.generalNameInputV2().clear().type(name);
+  e2e.components.DataSourcePicker.inputV2().type(`${dataSourceName}{enter}`);
+  e2eSelectors.variableEditor.queryType
+    .input()
+    .find('input')
+    .type(`${type.replace('Azure', '').trim()}{enter}`);
+  switch (type) {
+    case AzureQueryType.ResourceGroupsQuery:
+      e2eSelectors.variableEditor.subscription.input().find('input').type(`${options?.subscription}{enter}`);
+      break;
+    case AzureQueryType.NamespacesQuery:
+      e2eSelectors.variableEditor.subscription.input().find('input').type(`${options?.subscription}{enter}`);
+      e2eSelectors.variableEditor.resourceGroup.input().find('input').type(`${options?.resourceGroup}{enter}`);
+      break;
+    case AzureQueryType.ResourceNamesQuery:
+      e2eSelectors.variableEditor.subscription.input().find('input').type(`${options?.subscription}{enter}`);
+      e2eSelectors.variableEditor.resourceGroup.input().find('input').type(`${options?.resourceGroup}{enter}`);
+      e2eSelectors.variableEditor.namespace.input().find('input').type(`${options?.namespace}{enter}`);
+      break;
+    case AzureQueryType.MetricNamesQuery:
+      e2eSelectors.variableEditor.subscription.input().find('input').type(`${options?.subscription}{enter}`);
+      e2eSelectors.variableEditor.resourceGroup.input().find('input').type(`${options?.resourceGroup}{enter}`);
+      e2eSelectors.variableEditor.namespace.input().find('input').type(`${options?.namespace}{enter}`);
+      e2eSelectors.variableEditor.resource.input().find('input').type(`${options?.resource}{enter}`);
+      break;
+  }
+  e2e.pages.Dashboard.Settings.Variables.Edit.General.submitButton().click();
+  e2e.components.PageToolbar.item('Go Back').click();
+};
+
+e2e.scenario({
+  describeName: 'Add Azure Monitor datasource',
+  itName: 'fills out datasource connection configuration',
+  scenario: () => {
+    // This variable will be set in CI
+    const CI = e2e.env('CI');
+    if (CI) {
+      e2e()
+        .readFile('../../outputs.json')
+        .then((outputs) => {
+          provisionAzureMonitorDatasources([
+            {
+              datasources: [
+                {
+                  jsonData: {
+                    cloudName: 'Azure',
+                    tenantId: outputs.tenantId,
+                    clientId: outputs.clientId,
+                  },
+                  secureJsonData: { clientSecret: outputs.clientSecret },
+                },
+              ],
+            },
+          ]);
+        });
+    } else {
+      e2e()
+        .readFile(provisioningPath)
+        .then((azMonitorProvision: string) => {
+          const yaml = load(azMonitorProvision) as AzureMonitorProvision;
```

Comment:
can we use an eslint-ignore comment instead of adding an exception to betterer?

The plan is for us to remove betterer completely once we fix all the existing lint violations, but we can't do that if we keep adding new stuff to it!

Reply:
The issue is using `as AzureMonitorProvision` right here. That's where the comment should be added not anything in the e2e package

Reply:
You're correct, I was confused by the `any` returned in the Grafana E2E package, I've updated this and it should pass.